website/files/models/osfstorage.py

```diff
@@ -108,14 +108,15 @@ def is_checked_out(self):
         return self.checkout is not None
 
     def delete(self, user=None, parent=None):
-        if self.node.preprint_file == self:
-            self.node._is_preprint_orphan = True
-            self.node.save()
+        if self.node.preprint_file == self and not self.node._has_abandoned_preprint:
```

Comment:
We should have a property named something like "delete_allowed" that does the calculation here. That way we can serialize it and make it available to Fangorn and the API. Then this method would just be `if self.delete_allowed:`